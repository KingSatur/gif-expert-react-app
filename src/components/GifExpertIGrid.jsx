import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifExpertGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h1>{category}</h1>
      {isLoading && <h2>Loading...</h2>}
      <div className="card-grid">
        {images?.map((image) => {
          return <GifGridItem key={image?.id} {...image} />;
        })}
      </div>
    </>
  );
};

GifExpertGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
