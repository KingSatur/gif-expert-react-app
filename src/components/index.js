export * from "./AddCategory";
export * from "./GifExpertIGrid";
export * from "./GifGridItem";
