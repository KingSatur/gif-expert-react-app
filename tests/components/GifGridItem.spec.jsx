import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe("Test of <GifGridItem />", () => {
  const title = "dragon ball";
  const url = "drangonball.image";
  test("Should match with the snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show image with the url", () => {
    render(<GifGridItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toContain(url);
    expect(title).toBe(title);
  });

  test("Should show the title", () => {
    render(<GifGridItem title={title} url={url} />);
    const { textContent } = screen.getByText(title);
    expect(textContent).toContain(title);
  });
});
