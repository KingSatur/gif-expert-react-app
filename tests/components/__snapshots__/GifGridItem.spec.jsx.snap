// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <GifGridItem /> Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dragon ball"
      src="drangonball.image"
    />
    <p>
      dragon ball
    </p>
  </div>
</div>
`;
